### ISSUE TITLE
bug: ion-radio does not work with aria-label

### ISSUE THREAD

**Issue by NorthFred** [ionic-team/ionic-framework#27076]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

v7.x

### Current Behavior

Observed while migrating existing project from Ionic 6 to Ionic 7, with existing <ion-radio-group> with some <ion-radio> buttons:

```
[Ionic Warning]: ion-radio now requires providing a label with either the default slot or the "aria-label" attribute. To migrate, remove any usage of "ion-label" and pass the label text to either the component or the "aria-label" attribute.

Example: <ion-radio>Option Label</ion-radio>
Example with aria-label: <ion-radio aria-label="Option Label"></ion-radio>

Developers can use the "legacy" property to continue using the legacy form markup. This property will be removed in an upcoming major release of Ionic where this form control will use the modern form markup. 
```

When changing as suggested, e.g.:
```
        <ion-radio-group>
          <ion-radio aria-label="Label 1" slot="start"></ion-radio>
          <ion-radio aria-label="Label 2" slot="start"></ion-radio>
        </ion-radio-group>
```
only empty radio buttons are shown instead. The labels are missing.     

### Expected Behavior

Labels are shown.

### Steps to Reproduce

1. Create a radio button group.
2. Add some radio buttons.
3. Include labels for the radio buttons with 'aria-label'

### Ionic Info

Ionic:

   Ionic CLI                     : 7.0.0 (C:\Users\XXXXXXXXX\AppData\Roaming\npm\node_modules\@ionic\cli)
   Ionic Framework               : not installed
   @angular-devkit/build-angular : 15.2.4
   @angular-devkit/schematics    : 15.2.4
   @angular/cli                  : 15.2.4
   @ionic/angular-toolkit        : 9.0.0

Capacitor:

   Capacitor CLI      : 4.7.1
   @capacitor/android : 4.7.1
   @capacitor/core    : 4.7.1
   @capacitor/ios     : not installed

Cordova:

   Cordova CLI       : not installed
   Cordova Platforms : not available
   Cordova Plugins   : not available

Utility:

   cordova-res : 0.15.4
   native-run  : 1.7.1

System:

   Android SDK Tools : 26.1.1 (C:\Users\XXXXXXXXXX\AppData\Local\Android\Sdk)
   NodeJS            : v16.15.1 (C:\Program Files\nodejs\node.exe)
   npm               : 9.5.1
   OS                : Windows 11

**Comment by amandaejohnston** [comment]
Thank you for the issue. As with other HTML elements, using `aria-label` will not cause visible text to render. If you need a visible label, place the text in the default slot as shown in the first example:
```
<ion-radio-group>
  <ion-radio slot="start">Label 1</ion-radio>
  <ion-radio slot="start">Label 2</ion-radio>
</ion-radio-group>
```
Can you give this a try and let me know if it resolves your issue?

**Comment by NorthFred** [comment]
Issue resolved by suggested approach.
